src/marinade.ts

```diff
@@ -44,1311 +38,1339 @@ import {
   withRemoveVote,
   withUpdateVote,
 } from '@marinade.finance/directed-stake-sdk'
-import NodeWallet from '@coral-xyz/anchor/dist/cjs/nodewallet'
 import { withdrawStake } from '@solana/spl-stake-pool'
 import {
   computeExpectedSOL,
   identifyValidatorFromTx,
   selectSpecificValidator,
 } from './util/stake-pool-helpers'
 import {
+  Connection,
   Keypair,
   LAMPORTS_PER_SOL,
+  PublicKey,
   StakeProgram,
   Transaction,
 } from '@solana/web3.js'
-
-export class Marinade {
-  constructor(public readonly config: MarinadeConfig = new MarinadeConfig()) {}
-
-  readonly provider: Provider = new AnchorProvider(
-    this.config.connection,
-    new NodeWallet(web3.Keypair.generate()),
-    { commitment: 'confirmed' }
-  )
-
-  private directedStakeSdk = new DirectedStakeSdk({
-    connection: this.config.connection,
+import {
+  getValidatorRecords,
+  validatorDuplicationFlag,
+} from './marinade-state/marinade-state'
+import { MarinadeState } from './marinade-state/marinade-state.types'
+import { MarinadeProgramBuilders } from './programs/marinade-program-builders'
+import { isMarinadeReferralProgramBuilders } from './programs/marinade-referral-program'
+
+/**
+ * Constructs a DirectedStake contract SDK.
+ *
+ * @param connection - RPC connection to the Solana cluster
+ * @param walletPublicKey - the user wallet public key that the SDK works with as default value
+ * @returns directed stake sdk
+ */
+export function getDirectedStakeSdk({
```

Comment:
since I'm planning to add a file for native-stake methods, I think it might be nice to have the independent functions related to directed stake in a separate file as well, just to make this monster file slightly easier to navigate